apis/python/src/tiledbsoma/io/outgest.py

```diff
@@ -323,9 +323,26 @@ def to_anndata(
             "If X_layer_name is None, extra_X_layer_names must not be provided"
         )
 
+    # Problem: when the specified layer name does not exist.
```

Comment:
there is a pythonic way to handle this case, and I recommend we use it.

1. You create a module-level sentinal value called `MISSING` (or whatever you like)
2. The default `X_layer` value is this sentinal

For example:
```
class Sentinal:
   pass

MISSING = Sentinal()


def a_function(
    X_layer: str|None|Sentinal = MISSING
):
    if X_layer is MISSING:
        print("no value specified")
    elif X_layer is None:
        print("None was specified")
    else:
        print(f"{X_layer} was specified.")
```

You could also just skip the class declaration and assign `MISSING=object()`.  I did the class simply for clarity

Reply:
To be more explicit, this would let you equate `MISSING` with the `data` layer, and `None` with "do not read X layer" behavior.

Reply:
Thanks for the code-review feedback, @bkmartinjr !

Reply:
I'll use the pattern, which TIL (thank you!) but not the misspelling "Sentinal [sic]"

Reply:
TIL!